### ISSUE TITLE
Installation problem

### ISSUE THREAD

**Issue by haakym** [xthiago/pdf-version-converter#3]
Hi, thanks for this package. I am having some issues installing via composer in a Laravel project (5.3) just wondering how I can fix it. Am I unable to install if Laravel is using a newer version of Syfony?

My console output:
```
λ composer require xthiago/pdf-version-converter dev-master                                                                 
./composer.json has been updated                                                                                            
Loading composer repositories with package information                                                                      
Updating dependencies (including require-dev)                                                                               
Your requirements could not be resolved to an installable set of packages.                                                  
                                                                                                                            
  Problem 1                                                                                                                 
    - Installation request for xthiago/pdf-version-converter dev-master -> satisfiable by xthiago/pdf-version-conver        
ter[dev-master].                                                                                                            
    - Conclusion: remove symfony/process v3.1.5                                                                             
    - Conclusion: don't install symfony/process v3.1.5                                                                      
    - xthiago/pdf-version-converter dev-master requires symfony/process ~2.3 -> satisfiable by symfony/process[v2.3.        
0, v2.3.1, v2.3.10, v2.3.11, v2.3.12, v2.3.13, v2.3.14, v2.3.15, v2.3.16, v2.3.17, v2.3.18, v2.3.19, v2.3.2, v2.3.20        
, v2.3.21, v2.3.22, v2.3.23, v2.3.24, v2.3.25, v2.3.26, v2.3.27, v2.3.28, v2.3.29, v2.3.3, v2.3.30, v2.3.31, v2.3.32        
, v2.3.33, v2.3.34, v2.3.35, v2.3.36, v2.3.37, v2.3.38, v2.3.39, v2.3.4, v2.3.40, v2.3.41, v2.3.42, v2.3.5, v2.3.6,         
v2.3.7, v2.3.8, v2.3.9, v2.4.0, v2.4.1, v2.4.10, v2.4.2, v2.4.3, v2.4.4, v2.4.5, v2.4.6, v2.4.7, v2.4.8, v2.4.9, v2.        
5.0, v2.5.1, v2.5.10, v2.5.11, v2.5.12, v2.5.2, v2.5.3, v2.5.4, v2.5.5, v2.5.6, v2.5.7, v2.5.8, v2.5.9, v2.6.0, v2.6        
.1, v2.6.10, v2.6.11, v2.6.12, v2.6.13, v2.6.2, v2.6.3, v2.6.4, v2.6.5, v2.6.6, v2.6.7, v2.6.8, v2.6.9, v2.7.0, v2.7        
.1, v2.7.10, v2.7.11, v2.7.12, v2.7.13, v2.7.14, v2.7.15, v2.7.16, v2.7.17, v2.7.18, v2.7.19, v2.7.2, v2.7.20, v2.7.        
3, v2.7.4, v2.7.5, v2.7.6, v2.7.7, v2.7.8, v2.7.9, v2.8.0, v2.8.1, v2.8.10, v2.8.11, v2.8.12, v2.8.13, v2.8.2, v2.8.        
3, v2.8.4, v2.8.5, v2.8.6, v2.8.7, v2.8.8, v2.8.9].                                                                         
    - Can only install one of: symfony/process[v2.3.0, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.3.1, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.3.10, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.11, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.12, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.13, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.14, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.15, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.16, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.17, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.18, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.19, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.2, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.3.20, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.21, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.22, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.23, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.24, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.25, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.26, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.27, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.28, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.29, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.3, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.3.30, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.31, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.32, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.33, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.34, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.35, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.36, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.37, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.38, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.39, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.4, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.3.40, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.41, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.42, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.3.5, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.3.6, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.3.7, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.3.8, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.3.9, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.4.0, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.4.1, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.4.10, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.4.2, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.4.3, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.4.4, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.4.5, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.4.6, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.4.7, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.4.8, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.4.9, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.5.0, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.5.1, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.5.10, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.5.11, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.5.12, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.5.2, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.5.3, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.5.4, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.5.5, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.5.6, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.5.7, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.5.8, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.5.9, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.6.0, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.6.1, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.6.10, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.6.11, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.6.12, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.6.13, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.6.2, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.6.3, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.6.4, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.6.5, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.6.6, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.6.7, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.6.8, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.6.9, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.7.0, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.7.1, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.7.10, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.7.11, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.7.12, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.7.13, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.7.14, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.7.15, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.7.16, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.7.17, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.7.18, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.7.19, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.7.2, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.7.20, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.7.3, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.7.4, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.7.5, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.7.6, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.7.7, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.7.8, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.7.9, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.8.0, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.8.1, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.8.10, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.8.11, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.8.12, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.8.13, v3.1.5].                                                            
    - Can only install one of: symfony/process[v2.8.2, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.8.3, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.8.4, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.8.5, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.8.6, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.8.7, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.8.8, v3.1.5].                                                             
    - Can only install one of: symfony/process[v2.8.9, v3.1.5].                                                             
    - Installation request for symfony/process (locked at v3.1.5) -> satisfiable by symfony/process[v3.1.5].                
                                                                                                                            
                                                                                                                            
Installation failed, reverting ./composer.json to its original content.                                                     
```

**Comment by xthiago** [comment]
Hi,

It happens because the constraint I've put on composer.json file for `symfony/proccess` and `symfony/filesystem` packages requires version `2.3.x`. We can change it to accept newer versions but first we need to check if these dependencies has BC in new versions.

Alternatively, you can add an alias to the version of these dependencies in your `composer.json` in order to force composer to use the version that you already have.

```
# composer.json
"require": {
  "symfony/proccess": "version-you-have as 2.3.0",
  "symfony/filesystem": "version-you-have as 2.3.0",
  "xthiago/pdf-version-converter": "~1.0.x"
}
```

https://getcomposer.org/doc/articles/aliases.md#require-inline-alias

Please let me know if you have success or problems using the library with newer versions of dependencies.

**Comment by haakym** [comment]
@xthiago Thanks for your response and explanation.

I forked your project earlier and have changed the versions and I'll see how I get on. I'm guessing running the tests will reveal if anything is amiss?

Thanks again!

**Comment by haakym** [comment]
FYI

I'm using the following versions of symfony in a forked version of the package:

```
"symfony/filesystem": "~3.1",
"symfony/process": "~3.1"
```

And get the following in when running the tests:

```
λ phpunit -c tests                                                                                                                                                                                       
PHPUnit 4.5.1 by Sebastian Bergmann and contributors.                                                                                                                                                    

Configuration read from C:\...\pdf-version-converter\tests\phpunit.xml                                                                                                                           

..........................                                       26 / 26 (100%)                                                                                                                          

Time: 2.57 seconds, Memory: 4.00Mb
```

**Comment by zirikatzaile** [comment]
Hi , quite a long later, having upgraded from laravel 5.2 to 5.3 I can't get it working no way. I have done:

- composer require xthiago/pdf-version-converter dev-master
- After the fail, edit the downloaded vendor/xthiago/pdf-version-converter/composer.json to say:

`   
"symfony/filesystem": "3.1.*",
"symfony/process": "3.1.*"
`

 And then composer install, but still the error. So, desperate, sorry for the "robbery", I just move all the classes under my custom app/libraries and adapt their namespaces to be autoloaded. But, my trick gets me into a new surprising mess:


`local.ERROR: exception 'Symfony\Component\Debug\Exception\FatalErrorException' with message 'Class 'Symfony\Component\Filesystem\Filesystem' not found'`

What the hell?!?!? I do not really file such basic class in my new Laravel 5.3.  FGS !!!! :( :( :( :(

Please some help ....nearly 2 days with this (I hate upgrades)

**Comment by zirikatzaile** [comment]
Sorry for the last comment. I've managed my self to get it working just by:

- moving the xthiago classes to app/library/pdfutils and changing their namespaces
- installing required symfony components apart

Not standard, but too much time wasted up now

**Comment by xthiago** [comment]
Hello everyone,

The issues with different versions of symfony has been solved in release v1.0.5 of library by #16. Sorry for that inconvenience.

See the release for more details: https://github.com/xthiago/pdf-version-converter/releases/tag/v1.0.5
